tests/aggregation/tests.py

```diff
@@ -395,6 +395,25 @@ def test_count(self):
         vals = Book.objects.aggregate(Count("rating", distinct=True))
         self.assertEqual(vals, {"rating__count": 4})
 
+    def test_annotate_values_count(self):
+        vals = Book.objects.annotate(xprice=F('price')).filter(xprice__lte=30).values('publisher').distinct().annotate(count=Count('pk')).order_by('publisher')
```

Comment:
Please break this down into a couple of lines to make it easier to read. Also, the `distinct` call should be unnecessary for this bug, and only introduces extra work that distracts from the main problem.